src/device.rs

```diff
@@ -221,10 +205,34 @@ impl Device {
                 }
             };
 
-            Ok(Control { id, value })
+            Ok(Control { id: desc.id, value })
         }
     }
 
+    /// Convenience method to get the current control for an ID.
+    /// Requires an extra syscall compared to [`control`].
```

Comment:
The only reason the rustdoc build step doesn't fail is because this links to the `src/control.rs` module. Prefix it with `Device::` or `Self::` to get the right scoped symbol, and suffix with `()` to:

1. Make the reader aware that this link is pointing to _a function_;
2. Make rustdoc aware (and fail if not) that this points to a function.

```suggestion
    /// Requires an extra syscall compared to [`Device::control()`].
```